pkg/services/sqlstore/datasource.go

```diff
@@ -50,10 +50,12 @@ func (ss *SQLStore) getDataSource(ctx context.Context, query *datasources.GetDat
 func (ss *SQLStore) GetDataSources(ctx context.Context, query *datasources.GetDataSourcesQuery) error {
 	var sess *xorm.Session
 	return ss.WithDbSession(ctx, func(dbSess *DBSession) error {
-		if query.DataSourceLimit <= 0 {
-			sess = dbSess.Where("org_id=?", query.OrgId).Asc("name")
-		} else {
-			sess = dbSess.Limit(query.DataSourceLimit, 0).Where("org_id=?", query.OrgId).Asc("name")
+		sess = dbSess.Asc("name")
+		if query.OrgId > 0 {
```

Comment:
Risk allowing user querying all datasources in the database. **Shouldn't** happen, but maybe safer to introduce a new method GetAllDataSources?

Reply:
Sounds good! Will do!

Reply:
Done on https://github.com/grafana/grafana/pull/50463/commits/47748ab44d1337da9e427befaedd1c49a672b451 and https://github.com/grafana/grafana/pull/50463/commits/d5a0129cf23f18bd585af5e83939d205c3a499b5